Updated environment and config

ilyasoifer opened the pull request.

doron-st reviewed (approved):
oh-kay

ilyasoifer commented:
so much effort just to install gputils

mayalevy commented on ugvc/pipelines/sv_stats_collect.py:
```diff
@@ -0,0 +1,246 @@
+    p_args = parser.parse_args(args[1:])
+
+    sv_stats, concordance_stats, fp_stats_df = collect_sv_stats(p_args.svcall_vcf, p_args.concordance_h5)
+    results = {}
```
maybe this is clear, but maybe add how do we get such a concordance file? Or its required fields/ format

ilyasoifer replied:
Addressed

mayalevy commented on Dockerfile.jbvc:
```diff
@@ -35,9 +35,6 @@ RUN source ~/.bashrc && \
     pip install .
 
-#download cnvpytor additional necessary files (described here: https://github.com/abyzovlab/CNVpytor?tab=readme-ov-file#install-using-pip)
-RUN cnvpytor -download
```
Was removed on purpose?

ilyasoifer replied:
@TammushB - this caused issues with the build. do you need cnvpytor in ugvc?

TammushB replied:
we use it in ugbio-cnv docker. we can drop this from ugvc.

mayalevy reviewed (approved):
I think it looks good, I wonder why this code is located in VC, but I think I am a bit lost with all the arrangements in the repos, so maybe it is fine,

TammushB commented on test/unit/reports/test_sv_stats_collect.py:
```diff
@@ -0,0 +1,276 @@
+        df_calls = pd.DataFrame({"label": ["TP", "FP", "FP"], "qual": [0.95, 0.6, 0.5]})
+
+        # Mock `stats_utils.precision_recall_curve`
+        def mock_precision_recall_curve(gt, predictions, fn_mask, pos_label, min_class_counts_to_output):
```
declared twice: here and line255. 
maybe declare it in the class and not within the functions.

ilyasoifer replied:
fixed

TammushB commented on test/unit/reports/test_sv_stats_collect.py:
```diff
@@ -0,0 +1,276 @@
+        def mock_precision_recall_curve(gt, predictions, fn_mask, pos_label, min_class_counts_to_output):
+            return [0.8, 0.9], [0.7, 0.8], [0.5, 0.6], []
+
+        from ugbio_core import stats_utils
```
this import should be in the imports block

TammushB commented on test/unit/reports/test_sv_stats_collect.py:
```diff
@@ -0,0 +1,276 @@
+
+    def test_collect_sv_stats_without_concordance(self):
+        # Mock VCF data
+        vcf_data = """
```
used more than once. maybe declare it as a class variable

TammushB commented on ugvc/pipelines/sv_stats_collect.py:
```diff
@@ -0,0 +1,249 @@
+    gt = gt["label"]
+    gt = gt.replace({"FN": "TP"})
+    pos_label = "TP"
+    min_class_counts_to_output = 20
```
magic number

ilyasoifer replied:
converted to caps. Good enough?

TammushB reviewed (changes requested):
looks good. minor comments. 
maybe this is a hard/painful question, but why is SV related scripts/tools/reports are not ugbio-sv ?

ilyasoifer commented:
> looks good. minor comments. maybe this is a hard/painful question, but why is SV related scripts/tools/reports are not ugbio-sv ?

I would be happy to follow someone who moves those to ugbio_utils

ilyasoifer commented:
@TammushB - addressed your comments except for migrating this to ugbio. The issue is that it is all report infrastructure that needs to move. Don't have time to work on this now.
